tuf/src/client.rs

```diff
@@ -2238,4 +2265,91 @@ mod test {
             );
         })
     }
+
+    #[test]
+    fn client_can_update_with_unknown_len_and_hashes() {
```

Comment:
I may have missed it, but is there a test using defined lengths? If not, we should have one.

Reply:
Also, please fix the compiler/clippy errors below.